docs/docs/cloud/concepts/api.md

```diff
@@ -61,6 +61,98 @@ You can also specify multiple streaming modes at the same time. See the [how-to
+
+Streaming mode `events` is the same as using `.astream_events` in the LangGraph library - for a deeper conceptual explanation of this, you can see the LangGraph library documentation [here](../../concepts/low_level.md#streaming).
+
+#### `mode="messages"`
+Streaming mode `messages` is new streaming mode, currently only available in the API. What does this mode enable?
+
+This mode is focused on streaming back messages. It currently assumes that you have a `messages` key in your graph that is a list of messages. Assuming we have a simple react agent deployed, what does this stream look like?
+
+All events emitted have two attributes:
+
+- `event`: This is the name of the event
+- `data`: This is data associated with the event
+
+Let's run it on a question that should trigger a tool call:
+
+```python
+thread = await client.threads.create()
+input = {"messages": [{"role": "user", "content": "whats the weather in sf?"}]}
+
+events = []
+async for event in client.runs.stream(
+    thread["thread_id"],
+    assistant_id="agent",  # This may need to change depending on the graph you deployed
+    input=input,
+    stream_mode="messages",
+):
+    print(event.event)
+```
+```shell
+metadata
+messages/complete
+messages/metadata
+messages/partial
+...
+messages/partial
+messages/complete
+messages/complete
+messages/metadata
+messages/partial
+...
+messages/partial
+messages/complete
+end
+```
+
+We first get some `metadata` - this is metadata about the run. 
+
+```python
+StreamPart(event='metadata', data={'run_id': '1ef657cf-ae55-6f65-97d4-f4ed1dbdabc6'})
+```
+
+We then get a `messages/complete` event - this a fully formed message getting emitted. In this case,
+this was the just the input message we sent in. 
+
+```python
+StreamPart(event='messages/complete', data=[{'content': 'hi!', 'additional_kwargs': {}, 'response_metadata': {}, 'type': 'human', 'name': None, 'id': '833c09a3-bb19-46c9-81d9-1e5954ec5f92', 'example': False}])
+```
+
+We then get a `messages/metadata` - this is just letting us know that a new message is starting.
+
+```python
+StreamPart(event='messages/metadata', data={'run-985c0f14-9f43-40d4-a505-4637fc58e333': {'metadata': {'created_by': 'system', 'run_id': '1ef657de-7594-66df-8eb2-31518e4a1ee2', 'graph_id': 'agent', 'thread_id': 'c178eab5-e293-423c-8e7d-1d113ffe7cd9', 'model_name': 'openai', 'assistant_id': 'fe096781-5601-53d2-b2f6-0d3403f7e9ca', 'langgraph_step': 1, 'langgraph_node': 'agent', 'langgraph_triggers': ['start:agent'], 'langgraph_task_idx': 0, 'ls_provider': 'openai', 'ls_model_name': 'gpt-4o', 'ls_model_type': 'chat', 'ls_temperature': 0.0}}})
+```
+
+We then get a BUNCH of `messages/partial` events - these are the individual tokens from the LLM! In the case below, we can see the START of a tool call.
```

Comment:
maybe worth noting that this conceptually corresponds to LLM's `on_chat_model_stream`